### PR TITLE
Event data improvements

### PR DESCRIPTION
**Pull request by vdemedes**
This PR lays groundwork for #132 and #192 by allowing to easily send & receive custom test properties.

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -158,14 +158,15 @@ Runner.prototype._addTestResult = function (test) {
 		this.stats.failCount++;
 	}
 
-	this.results.push({
+	var props = {
 		duration: test.duration,
 		title: test.title,
-		error: test.assertError,
+		err: test.assertError,
```
Comment (sindresorhus) [inline lib/runner.js]: I would prefer `error` for consistency.

Comment (vdemedes) [inline lib/runner.js]: Haha, I changed it to `err` for consistency too :D

File: lib/runner.js [hunk 2]
```diff
@@ -158,14 +158,15 @@ Runner.prototype._addTestResult = function (test) {
...
 		type: test.type
-	});
+	};
 
-	this.emit('test', test.assertError, test.title, test.duration, test.type);
+	this.results.push(props);
+	this.emit('test', test.assertError, props);
```
Comment (sindresorhus) [inline lib/runner.js]: Why is `test.assertError` left in here? It's already part of `props`.

Comment (vdemedes) [inline lib/runner.js]: To conform to errback style. However, since it's the event, not a callback, I can skip it.

File: test/test.js [hunk 3]
```diff
@@ -405,10 +405,10 @@ test('runner have test event', function (t) {
 		a.end();
 	});
 
-	runner.on('test', function (err, title, duration) {
+	runner.on('test', function (err, props) {
 		t.error(err);
-		t.equal(title, 'foo');
-		t.ok(duration !== undefined);
+		t.equal(props.title, 'foo');
+		t.ok(props.duration !== undefined);
```
Comment (sindresorhus) [inline test/test.js]: This should use `t.not()`.

File: test/test.js [hunk 4]
```diff
@@ -405,10 +405,10 @@ test('runner have test event', function (t) {
 		a.end();
 	});
 
-	runner.on('test', function (err, title, duration) {
+	runner.on('test', function (err, props) {
 		t.error(err);
-		t.equal(title, 'foo');
-		t.ok(duration !== undefined);
+		t.equal(props.title, 'foo');
```
Comment (sindresorhus) [inline test/test.js]: Just fyi, tape supports `t.is` as an alias, so we could use that. Probably better as a separate PR though.

Comment (vdemedes) [inline test/test.js]: Yep, as part of that whole cleanup process.

File: lib/runner.js (hunk 1 above)
Comment (sindresorhus) [inline lib/runner.js]: I feel it's different. This is more like passing around data than variables. Think of it as a table of data and the keys are the headers. At least how it's like in my mind.

**Comment by sindresorhus** [comment]
Clean code is happy code :D

**Comment by vdemedes** [comment]
Couldn't agree more!

**Pull request merged**